runtime/lua/vim/provider/python.lua

```diff
@@ -0,0 +1,125 @@
+local M = {}
+
+--- @param prog string
+--- @param module string
+--- @return integer, string
+local function import_module(prog, module)
+  local program = [[
+import sys, importlib.util;
+sys.path = [p for p in sys.path if p != ""];
+sys.stdout.write(str(sys.version_info[0]) + "." + str(sys.version_info[1]));]]
+
+  program = program
+    .. string.format('sys.exit(2 * int(importlib.util.find_spec("%s") is None))', module)
+
+  local out = vim.system({ prog, '-W', 'ignore', '-c', program }):wait()
+  return out.code, assert(out.stdout)
+end
+
+local function get_python_candidates()
+  return {
+    'python3',
+    'python3.12',
+    'python3.11',
+    'python3.10',
+    'python3.9',
+    'python3.8',
+    'python3.7',
+    'python',
+  }
+end
+
+--- @param prog string
+--- @param module string
+--- @param major_version integer
+--- @return integer, string
+local function check_for_module(prog, module, major_version)
+  local prog_path = vim.fn.exepath(prog)
+  if prog_path == '' then
+    return 0, prog .. ' not found in search path or not executable.'
+  end
+
+  local min_version = '3.7'
```

Comment:
Maybe pull this out of the function to the top level? Would make it easier to change later on.